app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb

```diff
@@ -7,6 +7,100 @@ def available_vms
     availability_zone.vms.select { |vm| vm.format == volume_type }
   end
 
+  def self.params_for_create(ems)
+    {
+      :fields => [
+        {
+          :component  => 'text-field',
+          :name       => 'size',
+          :id         => 'size',
+          :label      => _('Size (in bytes)'),
+          :type       => 'number',
+          :step       => 1024 * 1024 * 1024,
+          :isRequired => true,
+          :validate   => [{:type => 'required'}],
+        },
+        {
+          :component => 'select',
+          :name      => 'volume_type',
+          :id        => 'volume_type',
+          :label     => _('Cloud Volume Type'),
+          :condition => {
+            :when => 'edit',
+            :is   => false,
+          },
+          :options   => ems.cloud_volume_types.map do |cvt|
+            {
+              :label => cvt.name,
+              :value => cvt.type,
+            }
+          end,
+        },
+        {
+          :component => 'switch',
+          :name      => 'multi_attachment',
+          :id        => 'multi_attachment',
+          :label     => _('Shareable'),
+          :onText    => _('Yes'),
+          :offText   => _('No'),
+        },
+        {
+          :component    => 'select',
+          :name         => 'affinity_policy',
+          :id           => 'affinity_policy',
+          :label        => _('Affinity Policy'),
+          :initialValue => 'off',
+          :condition    => {
+            :when => 'edit',
+            :is   => false,
+          },
+          :options      => [
+            {
+              :label => 'Off',
+              :value => 'off',
+            },
+            {
+              :label => 'Affinity',
+              :value => 'affinity',
+            },
+            {
+              :label => 'Anti-affinity',
+              :value => 'anti-affinity',
+            },
+          ],
+        },
+        {
+          :component  => 'select',
+          :name       => 'affinity_volume_id',
+          :id         => 'affinity_volume_id',
+          :label      => _('Affinity Volume'),
+          :isRequired => true,
+          :validate   => [{:type => 'required'}],
+          :condition  => {
+            :and => [
+              {
+                :not => {
+                  :when => 'affinity_policy',
+                  :is   => 'affinity',
```

Comment:
Found an error in the original schema I gave you @skateman 

This should be: `:is   => 'off',`. An 'affinity volume' is only applicable when `affinity` or `anti-affinity` was selected.

Reply:
:wrench: :hourglass: